projects/packages/account-protection/src/class-password-manager.php

```diff
@@ -95,12 +95,16 @@ public function validate_password_reset( \WP_Error $errors, $user ): void {
 	/**
 	 * Handle the profile update.
 	 *
-	 * @param int      $user_id The user ID.
-	 * @param \WP_User $old_user_data Object containing user data prior to update.
+	 * @param int                $user_id The user ID.
+	 * @param \WP_User|\stdClass $old_user_data Object containing user data prior to update.
 	 *
 	 * @return void
 	 */
-	public function on_profile_update( int $user_id, \WP_User $old_user_data ): void {
+	public function on_profile_update( int $user_id, $old_user_data ): void {
+		if ( ! is_object( $old_user_data ) || ! isset( $old_user_data->user_pass ) || '' === $old_user_data->user_pass ) {
```

Comment:
I just refactored a bit to use a guard clause and added an is_object() clause as well. Thoughts? @macbre

Reply:
I've become a huge fan of the `??` operator when it got introduced in PHP :-) But `is_object` should do it as well.

One could even simplify that condition to:

```suggestion
		if ( '' === ( $old_user_data->user_pass ?? '' ) ) {
```

Reply:
Gotcha. I think your approach will also require an `is_object()` check, since property access happens before null coalescing. Even if the method isn’t expected to receive anything other than an object, I can’t control what other plugins or themes might do, so I still need to ensure we’re receiving an object that includes the `user_pass` property.

With that in mind, I simplified the guard clause a bit by using `is_object()` along with `empty()`: https://github.com/Automattic/jetpack/pull/43045/commits/2a437ce671302a6921569adfdde34a80480abfd5

Thanks for the tip! The `??` operator is something I’d love to play around with more in PHP :)

Reply:
> I think your approach will also require an `is_object()` check, since property access happens before null coalescing

Not really, that's why the null coalescing operator is so handy:

```php
> $old_user_data = null;
> var_dump( $old_user_data->user_pass ?? '' );
string(0) ""
```

Reply:
Hmmm, I think something else was broken when I tested it then 🥲.
Sounds good, I'll keep it in mind next time, thanks!